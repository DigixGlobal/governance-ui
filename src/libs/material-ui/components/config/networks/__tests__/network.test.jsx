import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Network from '../network';

const network = {
  name: 'testNetwork',
  color: 'blue',
  image: 'test.png',
  description: 'test description',
};

describe('<Network />', () => {
  test('renders correctly', () => {
    const component = shallow(
      <Network
        web3={{}}
        updateNetwork={jest.fn}
        deleteNetwork={jest.fn}
        network={network}
      />,
    );

    expect(toJson(component)).toMatchSnapshot();
  });
});
