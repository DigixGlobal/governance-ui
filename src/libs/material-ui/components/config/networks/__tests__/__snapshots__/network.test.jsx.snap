// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Network /> renders correctly 1`] = `
<EZModal
  content={[Function]}
  data={
    Object {
      "color": "blue",
      "description": "test description",
      "image": "test.png",
      "name": "testNetwork",
    }
  }
  handleRemove={[Function]}
  handleSubmit={[Function]}
  header="Edit Network - testNetwork"
  trigger={
    <Segment
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      <Label
        basic={true}
        color="blue"
        image={true}
        ribbon={true}
      >
        <img
          alt=""
          src="test.png"
        />
        testNetwork
      </Label>
      <Header
        as="span"
        disabled={true}
        size="tiny"
      >
        test description
      </Header>
      <NetworkStatus
        name={false}
        network={
          Object {
            "color": "blue",
            "description": "test description",
            "image": "test.png",
            "name": "testNetwork",
          }
        }
        web3={Object {}}
      />
      <Segment />
    </Segment>
  }
/>
`;
